hwsecurity/fido2/src/main/java/de/cotech/hw/fido2/internal/pinauth/PinProtocol.java

```diff
@@ -0,0 +1,20 @@
+package de.cotech.hw.fido2.internal.pinauth;
+
+import java.io.IOException;
+
+import de.cotech.hw.fido2.internal.Fido2AppletConnection;
+
+public interface PinProtocol {
```

Comment:
The changes here aren't, strictly speaking, necessary for hmac-secret support. But if you ever want to implement the vastly superior CTAP2.1 PIN Protocol 2, you're going to need changes like these to do so.

I've used the CTAP2.1 terminology for the PIN operations - authenticate, encrypt, decrypt - instead of the implementation-specific ones that were here before.